packages/babel-helpers/src/helpers.js

```diff
@@ -461,6 +461,8 @@ helpers.set = template(`
 
       if (parent !== null) {
         set(parent, property, value, receiver);
+      } else {
+        receiver[property] = value;
       }
     } else if ("value" in desc && desc.writable) {
       desc.value = value;
```

Comment:
I _think_ the right way to solve it is also to replace it with basically `receiver[property] = value`. This will also execute the `OrdinarySet` algorithm. However, there needs to be a check for `'value' in Object.getOwnPropertyDescriptor(receiver, property)` (if there is a pre-existing descriptor). It needs to fail if the `receiver[property]` is a setter.